File: src/main/java/assets/ui/BackgroundAsset.java
package assets.ui;

import java.awt.Image;
import java.awt.image.BufferedImage;
import java.io.IOException;

import javax.imageio.ImageIO;

import assets.AssetHandle;

public class BackgroundAsset extends AssetHandle
{
	public BufferedImage bimg;

	public BackgroundAsset(AssetHandle asset)
	{
		super(asset);

		try {
			bimg = ImageIO.read(asset);
		}
		catch (IOException e) {
			bimg = null;
		}
	}

	@Override
	protected Image loadThumbnail()
	{
		return bimg;
	}
}

File: src/main/java/assets/ui/BackgroundAsset.kt
package assets.ui

import assets.Asset
import assets.AssetsDir
import java.awt.Image
import java.awt.image.BufferedImage
import java.io.IOException
import java.nio.file.Path
import javax.imageio.ImageIO

class BackgroundAsset(assetsDir: AssetsDir, relativePath: Path) : Asset(assetsDir, relativePath) {
	@JvmField
	val bimg: BufferedImage?

	init {
		bimg = try {
			ImageIO.read(getFile())
		} catch (e: IOException) {
			null
		}
	}

	override fun loadThumbnail(): Image? = bimg
}
